public/app/plugins/datasource/tempo/metricsSummary.ts

```diff
@@ -0,0 +1,294 @@
+    frame = emptyResponse;
+  } else if (state === LoadingState.Loading) {
+    frame = new MutableDataFrame({
+      name: 'Metrics Summary',
+      refId: 'metrics-summary',
+      fields: [
+        { name: 'state', type: FieldType.string, config: { displayNameFromDS: 'State', custom: { width: 300 } } },
+      ],
+      meta: {
+        preferredVisualisationType: 'table',
+      },
+    });
+    frame.add({
+      state: 'Loading...',
+    });
+  } else if (state === LoadingState.Done && data?.length) {
+    const dynamicMetrics: Record<string, FieldDTO> = {};
+    data.map((res: MetricsSummary) => {
+      const configQuery = getConfigQuery(res.series, targetQuery);
+      res.series.map((series: Series) => {
+        return !dynamicMetrics[series.key]
+          ? (dynamicMetrics[series.key] = {
+              name: `${series.key}`,
+              type: FieldType.string,
+              config: getConfig(series, configQuery, instanceSettings),
+            })
+          : dynamicMetrics[series.key];
+      });
+    });
+
+    frame = new MutableDataFrame({
+      name: 'Metrics Summary',
+      refId: 'metrics-summary',
+      fields: [
+        ...Object.values(dynamicMetrics).sort((a, b) => a.name.localeCompare(b.name)),
+        {
+          name: 'kind',
+          type: FieldType.string,
+          config: { displayNameFromDS: 'Kind', custom: { width: 150 } },
+        },
+        {
+          name: 'spanCount',
+          type: FieldType.string,
+          config: { displayNameFromDS: 'Span count', custom: { width: 150 } },
+        },
+        {
+          name: 'errorPercentage',
+          type: FieldType.string,
+          config: { displayNameFromDS: 'Error', unit: 'percent', custom: { width: 150 } },
+        },
+        getPercentileRow('p50'),
+        getPercentileRow('p90'),
+        getPercentileRow('p95'),
+        getPercentileRow('p99'),
+      ],
+      meta: {
+        preferredVisualisationType: 'table',
+      },
+    });
+
+    if (!data?.length) {
+      return emptyResponse;
+    }
+
+    const metricsData = data.map(transformToMetricsData);
+    for (const trace of metricsData) {
+      frame.add(trace);
+    }
+    frame = sortDataFrame(frame, 0);
+  }
+
+  if (!frame) {
+    frame = emptyResponse;
+  }
+
+  return [frame];
+}
+
+const transformToMetricsData = (data: MetricsSummary) => {
+  const errorPercentage = data.errorSpanCount
+    ? ((parseInt(data.errorSpanCount, 10) / parseInt(data.spanCount, 10)) * 100).toString()
+    : '';
+
+  const metricsData: MetricsData = {
+    kind: 'server', // so the user knows all results are of kind = server
+    spanCount: data.spanCount,
+    errorPercentage,
+    p50: data.p50,
+    p90: data.p90,
+    p95: data.p95,
+    p99: data.p99,
+  };
+
+  data.series.map((series: Series) => {
+    metricsData[`${series.key}`] = getMetricValue(series) || '';
+  });
+
+  return metricsData;
+};
+
+const getConfigQuery = (series: Series[], targetQuery: string) => {
```

Comment:
What does this actually do? I mean where does the `query` show in Grafana when returned like this? First time I am seeing it used so am curious.

Reply:
This is the query that is opened in explore when the user clicks on a dynamic attribute cell in the table.

Reply:
Sorry still don't get it. I mean don't we add data links to the frame to make the click work?

Reply:
This method constructs the query that goes in the link, so IIUC then yes.

![Screenshot 2023-08-23 at 10 52 38](https://github.com/grafana/grafana/assets/90795735/1bfe90fa-6a61-4856-ba6c-c22f0a437e97)

![Screenshot 2023-08-23 at 10 52 26](https://github.com/grafana/grafana/assets/90795735/1f82db12-9e0e-4855-ba41-7d7eaf9e7beb)

Reply:
Ah ok sorry got confused, thought this was some separate dataFrame config for some other purpose.

Reply:
No prob!